### PR TITLE
Adds vehicle spawns back to bridges

### LINKED ISSUE #68504: Vehicle wrecks on bridges stopped spawning at some point
### Describe the bug

Just a self reminder to look into and fix this

### Attach save file

N/A

### Steps to reproduce

1. Walk to a bridge
2. See no vehicle
3. :c

### Expected behavior

1. Drive to a bridge
2. See a cluster of vehicles blocking the bridge
3. \>:c

### Screenshots

_No response_

### Versions and configuration

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
Content "Vehicles spawn on bridges"

#### Purpose of change

Fixes #68504

#### Describe the solution

Country vehicle spawns also spawn on bridges
A single unique spawn where a truck veers into the barrier before crashing resulting in a small hole in the bridge
Minor changes to bridges to allow supports to function in multi z deep water (I thought bridges could spawn over lakes but at least the codes there for when they do)
Makes bridges use predecessor terrain so that once my JSON river PR is merged they'll gel nicely where say the river bank runs parallel to the bridge on one side.
Diagonal roads correctly join to bridges
Adds an EOC to change the weight of vanilla bridges so that Aftershock Exoplanet can disable them

#### Describe alternatives you've considered

Adding more special spawns
Adding boats that have hit or are tied to the supports?

#### Testing

Teled to a bunch of bridges
Checked the exoplanet EOC correctly disables the vanilla variants

### CODE REVIEW

File: data/json/overmap/overmap_terrain/overmap_terrain_transportation.json [hunk 1]
```diff
@@ -82,6 +76,13 @@
...
     "sym": "^"
   },
+  {
+    "type": "overmap_terrain",
+    "id": "godco_enter",
+    "name": "dirt road",
+    "sym": "│",
+    "copy-from": "generic_rural_road"
+  },
   {
```
Comment (Procyonae) [inline data/json/overmap/overmap_terrain/overmap_terrain_transportation.json:86]:
```suggestion
  },
  {
```

**Comment by CoroNaut** [comment]
Can vehicles spawn directly up or down from the ramp on either side where they might be invisible to the player? Experimental 3d would probably allow you to see the cars before you crash into them, but it still has bugs.

**Comment by Procyonae** [comment]
Yes, I could easily adjust the spawns to not be at the edge of the z1 ramp if that's desired tho

**Comment by CoroNaut** [comment]
Vehicles are hard to repair and going onto a bridge only to immediately crash would be pretty annoying. Personally I'd prefer it if they didn't spawn within maybe 5-8ish tiles from the ramp. That gives enough time to stop if the player is going ~30km/h. Of course the specifics are up to you, If 3D vision becomes a default feature, vehicles spawning on the ram would be perfectly fine.

**Comment by Procyonae** [comment]
That's fair, I'll just wait for the tests to run and then change it

**Comment by Procyonae** [comment]
I don't understand the aftershock exoplanet fail here, the mod overrides both the OMT and the map but moans about expecting a predecessor

**Comment by Maleclypse** [comment]
Doesn’t it say it’s missing a predecessor flag?  I’m not sure it’s specifically an aftershock failure? @John-Candlebury

**Comment by John-Candlebury** [comment]
No idea it's worked for years before without issue so its certainly something g here that is the problem.

**Comment by Procyonae** [comment]
I mean it seems to be flagging the error incorrectly, it's saying the map expects a predecessor and that the OMT doesn't specify it should but the exoplanet map doesn't require one (it should be overriding my map) and the OMT therefore shouldn't need the REQUIRES_PREDECESSOR flag (and it doesn't bc again it's overwritten by the exoplanet OMT)

**Comment by Procyonae** [comment]
Oh I might've been being dumb is it bc the exoplanet maps don't actually override the vanilla maps they just have such high weights that in practice the vanilla one shouldn't spawn?

**Comment by Maleclypse** [comment]
Probably

C

**Comment by Procyonae** [comment]
Once this is merged I'll expand the EOC stuff to aftershock roads and anything else that currently is using high weights in place of actually overriding their vanilla counterparts

**Comment by Procyonae** [comment]
Huh I really weren't expecting that to work

**Pull request merged**